src/main/java/io/quarkus/registry/app/maven/PlatformsContentProvider.java

```diff
@@ -40,7 +40,12 @@ public boolean supports(ArtifactCoords artifact, UriInfo uriInfo) {
     @Override
     public Response provide(ArtifactCoords artifact, UriInfo uriInfo) throws Exception {
         var quarkusVersion = artifact.getClassifier();
-        PlatformCatalog platformCatalog = registryClient.resolveCurrentPlatformsCatalog(quarkusVersion);
+        final PlatformCatalog platformCatalog;
+        if (quarkusVersion.equals("all")) {
```

Comment:
quarkusVersion may be null, better invert the condition here

Reply:
No, as a classifier, it's never null.

Reply:
I see, but actually these constructors are not testing against nulls on the classifier part, maybe it would be nice to fix that in the registry client PR:

```java
   protected ArtifactCoords(String[] parts) {
      this.groupId = parts[0];
      this.artifactId = parts[1];
      this.classifier = parts[2];
      this.type = parts[3] == null ? "jar" : parts[3];
      this.version = parts[4];
   }

   public ArtifactCoords(ArtifactKey key, String version) {
      this.key = key;
      this.groupId = key.getGroupId();
      this.artifactId = key.getArtifactId();
      this.classifier = key.getClassifier();
      this.type = key.getType();
      this.version = version;
   }
```

Reply:
Good point, thanks!